docs/design/queue.md

```diff
@@ -0,0 +1,111 @@
+    Aborted int32 `json:"aborted,omitempty" protobuf:"bytes,6,opt,name=aborted"`
+}
+```
+
+### QueueController
+
+The `QueueController` will manage the lifecycle of queue: 
+
+1. Watching `PodGroup`/`Job` for status
+2. If `Queue` was deleted, also delete all reated `PodGroup`/`Job` in the queue
+
+### Admission Controller
+
+The admission controller will check `PodGroup`/`Job` 's queue when creation:
+
+1. if the queue does not exist, the creation will be rejected
+2. if the queue is releasing, the creation will be also rejected
+
+### Feature Interaction
+
+#### Customized Job/PodGroup
+
+If the `PodGroup` is created by customized controller, the `QueueController` will count those `PodGroup` into `Unknown` status; because `PodGroup` focus on scheduling specification which did not include customized job's status. 
+
+#### cli
+
+Command line is also enhanced for operater engineers. Three sub-commands are introduced as follow:
+
+__create__:
+
+`create` command is used to create a queue with weight; for example, the following command will create a queue named `myqueue` with weight 10.
+
+```shell
+$ vkctl queue create --name myqueue --weight 10
+```
+
+__view__:
+
+`view` command is used to show the detail of a queue, e.g. creation time; the following command will show the detail of queue `myqueue`
+
+```shell
+$ vkctl queue view myqueue
+```
+
+__list__:
+
+`list` command is used to show all available queue to current user
+
+```shell
+$ vkctl queue list
+Name      Weight  Total  Pending  Running ...
+myqueue   10      10     5        5
+```
+
+#### Security
+
+TODO: who can view, use, delete/update related queues?
+
+#### Scheduler
+
+* Proportion plugin: 
+
+  Proportion plugin is used to share resource between `Queue`s by weight. The deserved resource of a queue is `(weight/total-weight) * total-resource`. When allocating resources, it will not allocate resource more than its deserved resources. 
+
+* Reclaim action: 
+
+  `reclaim` action will go through all queues to reclaim others by `ReclaimableFn`'s return value; the time complexity is `O(n^2)`. In `ReclaimableFn`, both `proportion` and `gang` will take effect: 1. `proportion` makes sure the queue will not be under-used after reclaim, 2. `gang` makes sure the job will not be reclaimed if its `minAvailable` > 1.
+
+* Backfill action:
+
+  When `allocate` action assign resources to each queue, there's a case that ([kube-batch#492](<https://github.com/kubernetes-sigs/kube-batch/issues/492>)) the resources maybe unnecessary idle because of `proportion` plugin: there are one pending job in two queue each, and the deserved resources of each queue can not meet the requirement of their jobs. In such case, `backfill` action will ignore deserved guarantee of queue to fill idle resources as much as possible. This introduces another potential case that the coming smaller job is blocked; this case will be handle be reserved resources of each queue in other project.
```

Comment:
`` this case will be handle be reserved resources of each queue in other project.`` is a little hard to understand for me:)

Reply:
should be `by reserved resource` :) we need to support reserved resource of each queue.